langserver/server.go

```diff
@@ -164,6 +188,45 @@ func (s *server) connectPrometheus(url string) error {
 			return err
 		}
 
+		// For prometheus version less than 2.14 `api/v1/status/buildinfo` was not supported this can
+		// break many function which solely depends on version comparing like `hover`, etc.
+		if resp.StatusCode == http.StatusNotFound {
+			err = errors.Errorf("Got %d when tried to access %s%s", resp.StatusCode, url, "/api/v1/status/buildinfo")
+
+			// nolint: errcheck
+			s.client.ShowMessage(s.lifetime, &protocol.ShowMessageParams{
+				Type:    protocol.Error,
+				Message: fmt.Sprintf("Failed to access build info via %s:\n\n%s ", url, err.Error()),
+			})
+
+			s.prometheus = nil
```

Comment:
In this case it could be a Prometheus of version < 2.14.0.

Since we still want to connect in that case lets instead assume we have a working Prometheus with version 2.13.

Additional checks whether the Prometheus is actually working can be introduced later.